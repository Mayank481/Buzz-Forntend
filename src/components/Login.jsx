import React, { useState } from 'react';
import { ToastContainer } from 'react-toastify';
import { handleGAuth, postLoginData } from '../services/authServices';

export default function Login({ fetchUser }) {
  const GoogleAuth = (e) => handleGAuth(e);
  const [inputs, setInputs] = useState({
    userEmail: '',
    userPassword: '',
  });

  const OnInputChange = (e) => {
    setInputs({ ...inputs, [e.target.name]: e.target.value });
  };

  const LocalAuth = (e) => postLoginData(e, fetchUser, inputs);

  return (
    <>
      <div className="">
        <div className="container ">
          <div className="d-flex justify-content-center align-items-center viewport-height">
            <div className=" d-flex justify-content-center align-items-center row shadow-lg p-3 mb-5 bg-body rounded p-5 w-100 login-panel  ">
              <div className="col-md-6">
                <div className="p-3">
                  <div className="d-flex justify-content-center">
                    <img
                      className="logo"
                      alt="logo"
                      src="https://media-exp1.licdn.com/dms/image/C560BAQGjUZbDAenjbw/company-logo_200_200/0/1519888826511?e=2147483647&v=beta&t=sM05vxlPsnwreOwKwVd3W0Jd4RKx9NK7uJrUMVI2Rtw"
                    />
                  </div>
                  <div className="text-center mt-3  font-family fw-bolder">
                    <h3>Enter your details and Start</h3>
                  </div>
                  <div className="text-center font-family ">
                    <h3>your journey with us</h3>
                  </div>
                  <div className="text-center font-family mt-5 text-muted">
                    Dont't stop you're proud
                  </div>
                  <div className="text-center mt-5">
                    <button
                    title="gAuthBtn"
                      onClick={GoogleAuth}
                      className="btn btn-lg google-button-clr rounded-pill  bg-transparent font-family font-size fw-bolder "
                    >
                      Sign In with Google
                    </button>
                  </div>
                </div>
              </div>
              <div className="col-md-6">
                <form method="POST">
                  <div className="p-3">
                    <h3 className=" text-center font-family fw-bolder">
                      Login To Your Account
                    </h3>
                    <input
                      type="email"
                      name="userEmail"
                      className="form-control border-0 border-bottom  mt-5"
                      onChange={(e) => OnInputChange(e)}
                      id="exampleFormControlInput1"
                      placeholder="TTN Email"
                      title="email-id"
                    />
                    <input
                      type="password"
                      className="form-control border-0 border-bottom   mt-5"
                      name="userPassword"
                      onChange={(e) => OnInputChange(e)}
                      id="exampleFormControlInput1"
                      placeholder="Password"
                      title="password"
                    />

                    <div className="text-center d-grid gap-1 mt-5">
                      <button
                      title="login-btn"
                      type="submit"
                        onClick={LocalAuth}
                        className="btn btn-lg px-5  mt-3 rounded-pill signup-btn-clr font-family fw-bolder font-size"
                      >
                        Sign In
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ToastContainer theme="colored" />
    </>
  );
}
