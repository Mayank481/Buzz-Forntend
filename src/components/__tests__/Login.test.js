import { render, screen, fireEvent } from '@testing-library/react';
import Login from "../Login";
import renderer from 'react-test-renderer';
test('should render Login Panel', () => {
  render(<Login />)
  const email_input = screen.queryByTitle('email-id');
  const pass_input = screen.queryByTitle('password');
  const loginBtn = screen.queryByTitle('login-btn');
  const gAuthBtn = screen.queryByTitle('gAuthBtn');
  expect(gAuthBtn).toBeTruthy();
  expect(email_input).toBeTruthy();
  expect(pass_input).toBeTruthy();
  expect(loginBtn).toBeTruthy();

  const component = renderer.create(<Login />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})
describe('Change login Fields Inputs', () => {
  it('onChange', () => {
    render(<Login />);
    const email_input = screen.queryByTitle('email-id');
    const pass_input = screen.queryByTitle('password');
    fireEvent.change(email_input, { target: { value: 'abc@tothenew.com' } });
    fireEvent.change(pass_input, { target: { value: 'abc123@com' } });
    expect(email_input.value).toBe('abc@tothenew.com');
    expect(pass_input.value).toBe('abc123@com');

    const component = renderer.create(<Login />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
  });
});