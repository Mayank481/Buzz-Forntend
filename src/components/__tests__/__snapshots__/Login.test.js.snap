// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change login Fields Inputs onChange 1`] = `
Array [
  <div
    className=""
  >
    <div
      className="container "
    >
      <div
        className="d-flex justify-content-center align-items-center viewport-height"
      >
        <div
          className=" d-flex justify-content-center align-items-center row shadow-lg p-3 mb-5 bg-body rounded p-5 w-100 login-panel  "
        >
          <div
            className="col-md-6"
          >
            <div
              className="p-3"
            >
              <div
                className="d-flex justify-content-center"
              >
                <img
                  alt="logo"
                  className="logo"
                  src="https://media-exp1.licdn.com/dms/image/C560BAQGjUZbDAenjbw/company-logo_200_200/0/1519888826511?e=2147483647&v=beta&t=sM05vxlPsnwreOwKwVd3W0Jd4RKx9NK7uJrUMVI2Rtw"
                />
              </div>
              <div
                className="text-center mt-3  font-family fw-bolder"
              >
                <h3>
                  Enter your details and Start
                </h3>
              </div>
              <div
                className="text-center font-family "
              >
                <h3>
                  your journey with us
                </h3>
              </div>
              <div
                className="text-center font-family mt-5 text-muted"
              >
                Dont't stop you're proud
              </div>
              <div
                className="text-center mt-5"
              >
                <button
                  className="btn btn-lg google-button-clr rounded-pill  bg-transparent font-family font-size fw-bolder "
                  onClick={[Function]}
                  title="gAuthBtn"
                >
                  Sign In with Google
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-md-6"
          >
            <form
              method="POST"
            >
              <div
                className="p-3"
              >
                <h3
                  className=" text-center font-family fw-bolder"
                >
                  Login To Your Account
                </h3>
                <input
                  className="form-control border-0 border-bottom  mt-5"
                  id="exampleFormControlInput1"
                  name="userEmail"
                  onChange={[Function]}
                  placeholder="TTN Email"
                  title="email-id"
                  type="email"
                />
                <input
                  className="form-control border-0 border-bottom   mt-5"
                  id="exampleFormControlInput1"
                  name="userPassword"
                  onChange={[Function]}
                  placeholder="Password"
                  title="password"
                  type="password"
                />
                <div
                  className="text-center d-grid gap-1 mt-5"
                >
                  <button
                    className="btn btn-lg px-5  mt-3 rounded-pill signup-btn-clr font-family fw-bolder font-size"
                    onClick={[Function]}
                    title="login-btn"
                    type="submit"
                  >
                    Sign In
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="Toastify"
  />,
]
`;

exports[`should render Login Panel 1`] = `
Array [
  <div
    className=""
  >
    <div
      className="container "
    >
      <div
        className="d-flex justify-content-center align-items-center viewport-height"
      >
        <div
          className=" d-flex justify-content-center align-items-center row shadow-lg p-3 mb-5 bg-body rounded p-5 w-100 login-panel  "
        >
          <div
            className="col-md-6"
          >
            <div
              className="p-3"
            >
              <div
                className="d-flex justify-content-center"
              >
                <img
                  alt="logo"
                  className="logo"
                  src="https://media-exp1.licdn.com/dms/image/C560BAQGjUZbDAenjbw/company-logo_200_200/0/1519888826511?e=2147483647&v=beta&t=sM05vxlPsnwreOwKwVd3W0Jd4RKx9NK7uJrUMVI2Rtw"
                />
              </div>
              <div
                className="text-center mt-3  font-family fw-bolder"
              >
                <h3>
                  Enter your details and Start
                </h3>
              </div>
              <div
                className="text-center font-family "
              >
                <h3>
                  your journey with us
                </h3>
              </div>
              <div
                className="text-center font-family mt-5 text-muted"
              >
                Dont't stop you're proud
              </div>
              <div
                className="text-center mt-5"
              >
                <button
                  className="btn btn-lg google-button-clr rounded-pill  bg-transparent font-family font-size fw-bolder "
                  onClick={[Function]}
                  title="gAuthBtn"
                >
                  Sign In with Google
                </button>
              </div>
            </div>
          </div>
          <div
            className="col-md-6"
          >
            <form
              method="POST"
            >
              <div
                className="p-3"
              >
                <h3
                  className=" text-center font-family fw-bolder"
                >
                  Login To Your Account
                </h3>
                <input
                  className="form-control border-0 border-bottom  mt-5"
                  id="exampleFormControlInput1"
                  name="userEmail"
                  onChange={[Function]}
                  placeholder="TTN Email"
                  title="email-id"
                  type="email"
                />
                <input
                  className="form-control border-0 border-bottom   mt-5"
                  id="exampleFormControlInput1"
                  name="userPassword"
                  onChange={[Function]}
                  placeholder="Password"
                  title="password"
                  type="password"
                />
                <div
                  className="text-center d-grid gap-1 mt-5"
                >
                  <button
                    className="btn btn-lg px-5  mt-3 rounded-pill signup-btn-clr font-family fw-bolder font-size"
                    onClick={[Function]}
                    title="login-btn"
                    type="submit"
                  >
                    Sign In
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="Toastify"
  />,
]
`;
